test/run_envoy_bazel_coverage.sh

```diff
@@ -18,32 +20,70 @@ if [[ $# -gt 0 ]]; then
 elif [[ -n "${COVERAGE_TARGET}" ]]; then
   COVERAGE_TARGETS=${COVERAGE_TARGET}
 else
-  COVERAGE_TARGETS=//test/...
+  # For fuzz builds, this overrides to just fuzz targets.
+  COVERAGE_TARGETS=//test/... && [[ ${FUZZ_COVERAGE} == "true" ]] &&
+    COVERAGE_TARGETS="$(bazel query 'attr("tags", "fuzz_target", //test/...)')"
 fi
 
-# Make sure //test/coverage:coverage_tests is up-to-date.
 SCRIPT_DIR="$(realpath "$(dirname "$0")")"
-"${SCRIPT_DIR}"/coverage/gen_build.sh ${COVERAGE_TARGETS}
+TEMP_CORPORA=""
+if [ "$FUZZ_COVERAGE" == "true" ]
+then
+  # Build and run libfuzzer linked target, grab collect temp directories.
+  FUZZ_TEMPDIR="$(mktemp -d)"
+  FUZZ_TEMPDIR=${FUZZ_TEMPDIR} "${SCRIPT_DIR}"/build_and_run_fuzz_targets.sh ${COVERAGE_TARGETS}
+else
+  # Make sure //test/coverage:coverage_tests is up-to-date.
+  "${SCRIPT_DIR}"/coverage/gen_build.sh ${COVERAGE_TARGETS}
+fi
+
+# Set the bazel targets to run.
+BAZEL_TARGET=//test/coverage:coverage_tests && [[ ${FUZZ_COVERAGE} == "true" ]] && BAZEL_TARGET=${COVERAGE_TARGETS}
 
 # Using GTEST_SHUFFLE here to workaround https://github.com/envoyproxy/envoy/issues/10108
-BAZEL_USE_LLVM_NATIVE_COVERAGE=1 GCOV=llvm-profdata bazel coverage ${BAZEL_BUILD_OPTIONS} \
+# Add binaries to OBJECTS to pass in to llvm-cov
+OBJECTS=""
+for t in ${BAZEL_TARGET}
+do
+  # Set test args. If normal coverage run, this is --log-path /dev/null
+  if [ "$FUZZ_COVERAGE" == "true" ]
+  then
+    # If this is a fuzz target, set args to be the temp corpus.
+    TARGET_BINARY="${t/://}"
+    CORPUS_LOCATION="${TARGET_BINARY:2}"
+    TEST_ARGS=(--test_arg="${FUZZ_TEMPDIR}/${CORPUS_LOCATION////_}_corpus" --test_arg="-runs=0")
+    if [[ -z "${OBJECTS}" ]]; then
+      # The first object needs to be passed without -object= flag.
+      OBJECTS="bazel-bin/${TARGET_BINARY:2}_with_libfuzzer"
+    else
+      OBJECTS="$OBJECTS -object=bazel-bin/${TARGET_BINARY:2}_with_libfuzzer"
+    fi
+    TARGET="${t}_with_libfuzzer"
+  else
+    TEST_ARGS=(--test_arg="--log-path /dev/null" --test_arg="-l trace")
+    OBJECTS="bazel-bin/test/coverage/coverage_tests"
```

Comment:
It feels like this should be outside the loop, as it's only done once.

Reply:
For clarity? In the case that it's just `coverage_tests` the loop only has one item.

Reply:
Yeah, maybe. I don't feel that strongly, it just took a somewhat higher cognitive hit processing that code in the loop :)

Reply:
Hm to resolve, I added a comment addressing the loop -- what I didn't want was to copy/paste the bazel command per loop, but if that makes more sense let me know!